main.c

```diff
@@ -36,14 +98,61 @@ void printHelp() {
 
     if (ptr != NULL)
         strcpy(selfName, ptr + 1);
-
     printf(HELP_TEXT, selfName, selfName);
 }
 
+
+
+void printVersion() {
+    // get self path
+    char selfName[PATH_MAX];
+#ifdef _WIN32
+    GetModuleFileName(NULL, selfName, PATH_MAX);
+    // get file name from path
+    const char *ptr = strrchr(selfName, '\\');
+#else
+    int ret = readlink("/proc/self/exe", selfName, sizeof(selfName) - 1);
+    if (ret == -1)
+        strcpy(selfName, "pak");
+    else
+        selfName[ret] = 0;
+    // get file name from path
+    const char *ptr = strrchr(selfName, '/');
+#endif
+
+    if (ptr != NULL)
+        strcpy(selfName, ptr + 1);
+	
+    printf(PAK_VERSION_STRING);

```

Comment:
1. puts and fputs is perferred.
2. The standalone switch for version should print sonething readable by machine, human-readable stuff can go with the `help` text.
3. Versions should be defined as macro so it can be easily updated when needed.